src/main/java/com/coveo/spillway/SpillwayFactory.java

```diff
@@ -58,9 +60,18 @@ public SpillwayFactory(LimitUsageStorage storage, Clock clock) {
    * @param resource The name of the resource on which the limit are enforced
    * @param limits The different enforced limits
    * @return The new {@link Spillway}
+   * @throws SpillwayLimitWithSameNameException
    */
   @SafeVarargs
-  public final <T> Spillway<T> enforce(String resource, Limit<T>... limits) {
+  public final <T> Spillway<T> enforce(String resource, Limit<T>... limits)
+      throws SpillwayLimitWithSameNameException {
+    int numberOfDistinctLimits =
+        (int) Arrays.stream(limits).map(limit -> limit.getName()).distinct().count();
+
+    if (numberOfDistinctLimits != Arrays.asList(limits).size()) {
+      throw new SpillwayLimitWithSameNameException();
+    }
```

Comment:
If we really want to throw on duplicate names, I vote we do it like this:

```
Arrays.stream(limits).collect(Collectors.groupBy(Limit::getName, Collectors.counting())
``` 
or something like that.

We get the actual problematic names.